src/database/__tests__/firestore.spec.ts

```diff
@@ -185,4 +155,186 @@ describe('firestore', () => {
+            const data = { mockData: 'data' }
+            const documentSnapshot = {
+                exists: true,
+                data: () => data
+            }
+            reference.get.mockReturnValue(Promise.resolve(documentSnapshot));
+
+            firestore.getDoc('collection', 'docId')
+                .subscribe(
+                    entity => {
+                        expect(entity).toBe(data);
+                        done();
+                    },
+                    err => {
+                        done.fail(new Error('Observable should not pass error'));
+                    }
+                );
+        });
+
+        it('passes an error to the observable when the document does not exist', done => {
+            const documentSnapshot = {
+                exists: false,
+                data: () => null
+            }
+            reference.get.mockReturnValue(Promise.resolve(documentSnapshot));
+
+            firestore.getDoc('collection', 'docId')
+                .subscribe(
+                    entity => {
+                        done.fail(new Error('Observable should not receive document snapshot'));
+                    },
+                    err => {
+                        expect(err.message).toBe('Document does not exist');
+                        done();
+                    }
+                );
+        });
+
+        it('passes the getDoc() error to the observable', done => {
+            const testErr = new Error();
+            reference.get.mockReturnValue(Promise.reject(testErr));
+
+            firestore.getDoc('collection', 'docId')
+                .subscribe(
+                    entity => {
+                        done.fail(new Error('Promise should not resolve'));
+                    },
+                    err => {
+                        expect(err).toBe(testErr);
+                        done();
+                    }
+                );
+        });
+    });
+
+    describe('deleteDoc()', () => {
+        it('successfully deletes a document', done => {
+            reference.delete.mockReturnValue(Promise.resolve());
+            firestore.deleteDoc('collection', 'docId')
+                .subscribe(
+                    () => {
+                        expect(reference.delete).toHaveBeenCalled();
+                        done();
+                    }
+                );
+
+        });
+
+        it('passes the getDoc() error to the observable', done => {
+            const testErr = new Error();
+            reference.delete.mockReturnValue(Promise.reject(testErr));
+
+            firestore.deleteDoc('collection', 'docId')
+                .subscribe(
+                    entity => {
+                        done.fail(new Error('Promise should not resolve'));
+                    },
+                    err => {
+                        expect(err).toBe(testErr);
+                        done();
+                    }
+                );
+        });
+    });
+
+    afterAll(() => {
```

Comment:
we don't need to close the connection for unit tests, instead just write unit tests for this function